src/main/java/com/keniu/models/Book.java

```diff
@@ -31,12 +42,11 @@ public class Book {
 
     /** The ISBN of the book, which should be unique. */
     @NotBlank
-    @Column(nullable = false)
-    // todo add unique annotation
+    @Column(nullable = false, unique = true)
     private String isbn;
 
     /** The price of the book. */
-    @NotBlank
+    @NotNull
```

Comment:
you can remove it. Better use validators with request dtos
```suggestion
```

Reply:
I think I can leave this. It's another layer of validation

Reply:
Yes, we should because they serve different purposes. They complement each other. I disagree with the statement "Both of these approaches do not allow us to store null values in the database." @ NotNull allows creating entity with null value in DB. @ NotNull helps quickly detect incorrect data if any arise. @ Column(nullable = false) sets a constraint at database level. For example, if our created object somehow changes and the value of a field becomes null, DB will not allow such invalid object to be created. Once again, they don't complicate code. They are for different purposes. They complement each other

Reply:
Don't you check for null values ​​at the dto level?

Reply:
`createBookRequestDto.setIsbn(null)` Should we save it?